docs/setup/aws.md

```diff
@@ -1,107 +1,148 @@
+They should be named according to the value of `TAG`.
 
-    Setting this enables [Watchful](https://github.com/eladb/cdk-watchful) for
-    monitoring Grapl with email alerts.
+### Configure CDK deployment parameters
 
-4. `GRAPL_CDK_OPERATIONAL_ALARMS_EMAIL` (optional)
+Grapl's CDK deployment parameters are set as environmental variables.
 
-    Setting this enables alarms meant for the operator of the Grapl stack.
+For a direct code reference, c.f. `src/js/grapl-cdk/bin/deployment_parameters.ts`.
 
-5. `GRAPL_CDK_SECURITY_ALARMS_EMAIL` (optional)
+#### Recommended approach
 
-    Setting this enables alarms meant for the consumer of the Grapl
-    stack, for example, "a new risk node has been found".
+If you are deploying to production, we recommend saving a `source`-able
+collection of these environment variables:
 
-When deploying to production we recommend creating a `source`-able 
-collection of these environment variables, and saving that to some
-version control; something along the lines of the following:
 ```bash
-export GRAPL_ROOT="~/src/grapl"
+# example values
 export GRAPL_DEPLOYMENT_NAME="some-grapl-deployment-name"
 export GRAPL_VERSION="latest"
+export GRAPL_ROOT="~/src/grapl"
 export GRAPL_REGION="us-west-2"
 export GRAPL_CDK_WATCHFUL_EMAIL="someone+watchful@domain.com"
 export GRAPL_CDK_OPERATIONAL_ALARMS_EMAIL="someone+alarm@domain.com"
 export GRAPL_CDK_SECURITY_ALARMS_EMAIL="someone+alarm@domain.com"
 export GRAPL_DGRAPH_INSTANCE_TYPE='i3.large'
 ```
 
-## Install graplctl
+#### Parameter explanation
+
+1. `GRAPL_DEPLOYMENT_NAME` (required)
+
+    A name for the deployment to AWS.  ([AWS naming requirements](https://docs.aws.amazon.com/awscloudtrail/latest/userguide/cloudtrail-s3-bucket-naming-requirements.html) apply)
+
+    This name must be globally (AWS) unique, as names for AWS S3 buckets will be dervied from this.
+
+    We recommend prefixing the deployment name with "Grapl-" to help identify Grapl resources in your AWS account.
+
+2. `GRAPL_VERSION` (required, if TAG changed)
+
+    Which locally built version of Grapl to deploy.
+
+    This string will be used to identify build outputs by their filenames in the `src/js/grapl-cdk/zips/` directory.
+
+    If you changed `TAG` earlier, you must set this to be that same value.
+
+    Otherwise, you may safely let it default to 'latest'.
+
+3. `GRAPL_ROOT` (required)
+
+    This is the path to the directory where you checked Grapl out as a Git repository.
+
+4. `GRAPL_REGION` (required)
+
+    This is your [AWS region](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html).
+
+5. `GRAPL_CDK_WATCHFUL_EMAIL` (optional)
+
+    Setting this enables [Watchful](https://github.com/eladb/cdk-watchful) for monitoring Grapl with email alerts.
+
+6. `GRAPL_CDK_OPERATIONAL_ALARMS_EMAIL` (optional)
+
+    Setting this enables alarms meant for the operator of the Grapl stack.
+
+7. `GRAPL_CDK_SECURITY_ALARMS_EMAIL` (optional)
```

Comment:
Are these email variables actually optional? I don't think they were last time I ran through this process.

Reply:
I remember having some difficulty with them too.  I'm unsure whether the documentation or the behavior reflects the intent.